packages/entity-server/src/routes/hierarchy/middleware/middleware.ts

```diff
@@ -34,16 +22,33 @@ export const attachContext = async (req: HierarchyRequest, res: Response, next:
 
     const hierarchy = await req.models.entityHierarchy.findOne({ name: hierarchyName });
     const entity = await req.models.entity.findOne({ code: entityCode });
+    if (!hierarchy || !entity) {
+      throwNoAccessError(hierarchyName, entityCode);
+    }
+
+    const rootEntity = entity.isProject()
+      ? entity
+      : await req.models.entity.findOne({ code: hierarchy.name });
```

Comment:
Hmmm - this would be adding an assumption/constraint that we don't currently have. I'd prefer to avoid binding projects <-> hierarchies in a one to one relationship. You might have e.g. a "Fiji Water Catchment Hierarchy" that is used by the "WISH Fiji" project and a future "Fiji Flood Risk" project.

Reply:
Wanna have a quick chat about this one? I actually considered this as I was writing the code but I guess my understanding is that we are already tied into that relationship due to there only being able to be one root entity per entity_hierarchy? I also feel it's not quite what we want, so would be happy to discuss how we can avoid this 😄

Reply:
Yep let's have a chat - I'll call you in 15